public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -94,8 +94,19 @@ export class PanelChrome extends PureComponent<Props, State> {
       if (data.state === LoadingState.Error) {
         const { error } = data;
         if (error) {
-          if (this.state.errorMessage !== error.message) {
-            this.setState({ errorMessage: error.message });
+          let message = 'Query error';
+          if (error.message) {
+            message = error.message;
+          } else if (error.data && error.data.message) {
+            message = error.data.message;
+          } else if (error.data && error.data.error) {
+            message = error.data.error;
+          } else if (error.status) {
+            message = `Query error: ${error.status} ${error.statusText}`;
+          }
```

Comment:
Interesting. Can't we unify this to one shape on query runner level?

Reply:
I mean at least when exposed from query runner. I guess the reason for DataQueryError is datasources returning different shapes for errors?

Reply:
For sure -- this is just copied from the existing error support.  I think if we know that the error is really a `DataQueryError` the support should be more clear